src/lib/storage/fixed_size_attribute_vector.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <vector>
+
+#include "base_attribute_vector.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+template <typename T>
+class FixedSizeAttributeVector : public BaseAttributeVector {
+ public:
+  explicit FixedSizeAttributeVector(const size_t size) { _attribute_vector = std::vector<T>(size); }
+
+  // returns the value id at a given position
+  ValueID get(const size_t i) const override { return ValueID(_attribute_vector.at(i)); }
+
+  // sets the value id at a given position
+  void set(const size_t i, const ValueID value_id) override { _attribute_vector[i] = value_id; }

```

Comment:
I'd think about using `_attribute_vector.at(i) = value_id;` here to make use of the error handling of the vector. Also makes the method more consistent with the get method